src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/SynchronisationService.kt

```diff
@@ -3,11 +3,12 @@ package uk.gov.justice.digital.hmpps.prisonertonomisupdate.services
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.module.kotlin.readValue
 import com.microsoft.applicationinsights.TelemetryClient
+import kotlin.reflect.full.memberProperties
 
 abstract class SynchronisationService(
```

Comment:
Shame this is abstract then you could let Spring wire up the dependencies. Can you make retryCreateMapping accept a lambda?

Reply:
Actually I don't think I need this anymore. The `tryCreateMapping` is no longer used since the helper class does all that. So all it needs is an interface which implements `suspend fun retryCreateMapping(message: String)`